src/index.js

```diff
@@ -49,7 +49,9 @@ class DelegatedPeerRouting {
     try {
       return await this._httpQueue.add(async () => {
         const { addrs } = await this.dht.findPeer(id, {
-          timeout: `${options.timeout}ms`// The api requires specification of the time unit (s/ms)
+          searchParams: {
+            timeout: `${options.timeout}ms`// The api requires specification of the time unit (s/ms)
+          }
```

Comment:
```suggestion
          timeout: options.timeout
```
i was looking into this in vasco's PR and the string timeout doesn't work correctly for some reason.

Reply:
Hmm, are you sure?  Against go-ipfs:

```console
$ time curl http://localhost:5001/api/v0/dag/get?arg=/ipfs/QmaA9foTB68jrGgP3dBcyc8nrjbfAji8xrJsrJYo22BiVo\&timeout=5s
{"Message":"failed to get block for QmaA9foTB68jrGgP3dBcyc8nrjbfAji8xrJsrJYo22BiVo: context deadline exceeded","Code":0,"Type":"error"}

real	0m5.044s
user	0m0.006s
sys	0m0.015

$ time curl http://localhost:5001/api/v0/dag/get?arg=/ipfs/QmaA9foTB68jrGgP3dBcyc8nrjbfAji8xrJsrJYo22BiVo\&timeout=5000

real	0m0.057s
user	0m0.007s
sys	0m0.014s

$ time curl http://localhost:5001/api/v0/dag/get?arg=/ipfs/QmaA9foTB68jrGgP3dBcyc8nrjbfAji8xrJsrJYo22BiVo\&timeout=1

real	0m0.048s
user	0m0.007s
sys	0m0.017s

$ ipfs dag get /ipfs/QmaA9foTB68jrGgP3dBcyc8nrjbfAji8xrJsrJYo22BiVo --timeout 1
Error: time: missing unit in duration 1
```

Reply:
i was talking about the client timeout, with string the tests timeout instantly with number they run.

Reply:
Context: https://github.com/libp2p/js-libp2p-delegated-peer-routing/pull/25

I did not re-try this yet with the fix hugo's recommended

Reply:
Context: https://github.com/libp2p/js-libp2p-delegated-peer-routing/pull/25

I did not re-try this yet with the fix hugo's recommended

Reply:
I think when these tests were written, we only had timeouts in a few HTTP method calls - the argument here is for the remote node, I think.

We could do both and set the timeout in the client and the server, but that will make it non-deterministic.

Reply:
Ideally we should perform set the timeout for the client and allow the cancel of that to cause the remote context to close. It sounds like there are some downstream issues with this at the moment though, so for now we can use the remote timeout to get around this until context cancelling is working appropriately.

> We could do both and set the timeout in the client and the server

Definitely want to avoid that, it would be a nightmare to deal with.